stat/stat.go

```diff
@@ -0,0 +1,59 @@
+package stat
+
+import (
+	"sync"
+	"time"
+)
+
+type Stat struct {
+	// requestTimes is a map of hour -> method -> count
+	requestTimes map[int]map[string]int
+	// initTimes is a map of hour -> time
+	initTimes  map[int]time.Time
+	requestsMu *sync.Mutex
+}
+
+// day is the number of nanoseconds in a day
+const day = 24 * time.Hour
+
+func NewStat() Stat {
+	requestTimes := make(map[int]map[string]int)
+	initTimes := make(map[int]time.Time)
+	return Stat{
+		requestTimes: requestTimes,
+		initTimes:    initTimes,
+		requestsMu:   &sync.Mutex{},
+	}
+}
+
+func (stat *Stat) Get() map[string]int {
+	t := time.Now()
+	numRequests := make(map[string]int)
+	stat.requestsMu.Lock()
+	for hour, methodTimestamps := range stat.requestTimes {
+		// only count the hour if it was within the past day
+		if t.Sub(stat.initTimes[hour]) > day {
+			continue
+		}
+
+		for method, count := range methodTimestamps {
+			numRequests[method] += count
```

Comment:
We are effectively copying the map here, so I think it is a little misleading to use `+=` rather than just assignment.
```suggestion
			numRequests[method] = count
```

Reply:
Actually just realised that this is summing up for the whole day using the one hour buckets, so this is actually correct, my bad.